packages/paste-core/components/code-block/__tests__/index.spec.tsx

```diff
@@ -0,0 +1,388 @@
+text "Hey there,\nPush a button!"
+background '#3333AA'
+foreground '#CCCCFF'
+}`}
+        </CodeBlock>
+      </Theme.Provider>
+    );
+
+    const codeBlock = screen.getByTestId('code-block');
+    const header = codeBlock.querySelector('[data-paste-element="CODE_BLOCK_HEADER"]');
+
+    expect(header).toBeDefined();
+    expect(codeBlock.getAttribute('aria-labelledby')).toBe(header?.id);
+
+    const headerLabel = screen.getByText('Ruby');
+    expect(headerLabel).toBeDefined();
+
+    const copyButton = screen.getByRole('button', {name: 'Copy code block'});
+    expect(copyButton).toBeDefined();
+  });
+
+  it('should render basic header if it is single line with wrapLines', () => {
+    render(
+      <Theme.Provider theme="default">
+        <CodeBlock language="javascript" data-testid="code-block" wrapLines>{`(num) => num + 1`}</CodeBlock>
+      </Theme.Provider>
+    );
+
+    const codeBlock = screen.getByTestId('code-block');
+    const header = codeBlock.querySelector('[data-paste-element="CODE_BLOCK_HEADER"]');
+
+    expect(header).toBeDefined();
+    expect(codeBlock.getAttribute('aria-labelledby')).toBe(header?.id);
+
+    const headerLabel = screen.getByText('JavaScript');
+    expect(headerLabel).toBeDefined();
+
+    const copyButton = screen.getByRole('button', {name: 'Copy code block'});
+    expect(copyButton).toBeDefined();
+  });
+
+  it('should render single line code block', () => {
+    render(
+      <Theme.Provider theme="default">
+        <CodeBlock
+          language="javascript"
+          data-testid="code-block"
+          externalLink="https://www.google.com"
+        >{`(num) => num + 1`}</CodeBlock>
+      </Theme.Provider>
+    );
+
+    const codeBlock = screen.getByTestId('code-block');
+    const header = codeBlock.querySelector('[data-paste-element="CODE_BLOCK_HEADER"]');
+    expect(header).toBeNull();
+
+    const copyButton = screen.getByRole('button', {name: 'Copy code block'});
+    expect(copyButton).toBeDefined();
+
+    const link = screen.getByRole('link', {name: 'Open code block in new page'});
+    expect(link).toBeDefined();
+  });
+
+  it('should render a link when pass externalLink prop', () => {
+    render(
+      <Theme.Provider theme="default">
+        <CodeBlock language="ruby" externalLink="https://www.google.com">
+          {`#!/usr/bin/ruby
+
+# Import the library. This is a really really really really really really really really really really really really long line that should be wrapped.
+require 'tk'
+
+# Root window.
+root = TkRoot.new  {
+title 'Push Me'
+background '#111188'
+}
+
+# Add a label to the root window.
+lab = TkLabel.new(root) {
+text "Hey there,\nPush a button!"
+background '#3333AA'
+foreground '#CCCCFF'
+}`}
+        </CodeBlock>
+      </Theme.Provider>
+    );
+
+    const link = screen.getByRole('link', {name: 'Open code block in new page'});
+    expect(link).toBeDefined();
+  });
+
+  it('should render the label when pass label prop', () => {
```

Comment:
suggestion (non-blocking): Try to merge some of these tests where the render is the same and you're checking a bunch of individual props. example: `it('handles label and externalLink props', () => {`.

Preventing duplicate renders, where it makes sense, speeds up test running